File: 0154-find-minimum-in-rotated-sorted-array-ii/Solution.cs
public class Solution {
    public int FindMin(int[] nums)
    {
        var left = 0;
        var right = nums.Length - 1;
        
        while (left < right)
        {
            if (nums[right] == nums[left])
            {
                right--;
                continue;
            }
            var mid = left + (right - left) / 2;
            if (nums[mid] > nums[right])
            {
                left = mid + 1;
            }
            else
            {
                right = mid;
            }
        }
        
        return nums[left];
    }
}
// [3,3,1,3]

File: 0154-find-minimum-in-rotated-sorted-array-ii/Solution.go
package Leetcode

func findMin(nums []int) int {
	left := 0
	right := len(nums) - 1
	for left < right {
		if nums[right] == nums[left] {
			right--
			continue
		}
		mid := left + (right-left)/2
		if nums[mid] > nums[right] {
			left = mid + 1
		} else {
			right = mid
		}
	}
	return nums[left]
}
